modules/core/src/main/scala/gem/parser/Angle.scala

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2016-2017 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package gem
+package parser
+
+import atto._, Atto._
+import gem.math.{ Angle, HourAngle }
+
+/** Parsers for `[[gem.math.Angle]].` */
+trait AngleParsers {
+  import MiscParsers._
+
+  /**
+   * Generic parser for the components of an angle in "11 22 33.444555" format, with at least 1
+   * and at most 6 digits following the decimal point, and terminal parsers for each segment.
+   */
+  def genAngle(t1: Parser[_], t2: Parser[_], t3: Parser[_]): Parser[(Int, Int, Int, Int, Int)] =
+    for {
+      h  <- int <~ t1
+      m  <- int <~ t2
+      s  <- int <~ char('.')
+      µs <- frac(6)
+      _  <- t3
+    } yield (h, m, s, µs / 1000, µs % 1000)
```

Comment:
Well, here I don't think using the applicative matters that much here. The `<*` `*>` `$>` `<$` can also come handy for parsing combinators but if they are such a pain to write I don't think it's worth bothering.

I wonder why the `|@|` doesn't have the lowest associative precedence, when would you ever need to have the `|@|` start associating from the other side? `<rant>`If the applicative syntax wasn't so cumbersome they could be actually used more generally`</rant>`.

I hope some kind of [syntactic sugar](https://downloads.haskell.org/~ghc/latest/docs/html/users_guide/glasgow_exts.html#applicative-do-notation) for Applicatives becomes more widespread.